qa/workunits/rbd/cli_generic.sh

```diff
@@ -1463,6 +1463,176 @@ test_mirror_snapshot_schedule_recovery() {
+            grep -c mirror.primary)" -gt '1' && break
+        sleep 10
+    done
+
+    test "$(rbd mirror group status rbd2/ns1/test1 |
+        grep -c mirror.primary)" -gt '1'
+
+    # repeat with kicked in schedule, see https://tracker.ceph.com/issues/53915
+    expect_fail rbd mirror group snapshot schedule ls
+    rbd mirror group snapshot schedule ls -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2
+    rbd mirror group snapshot schedule ls -p rbd2 -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2/ns1
+    rbd mirror group snapshot schedule ls -p rbd2/ns1 -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    test "$(rbd mirror group snapshot schedule ls -p rbd2/ns1 --group test1)" = 'every 1m'
+
+    rbd mirror group snapshot schedule status
+    test "$(rbd mirror group snapshot schedule status --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/test1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/test1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2/ns1 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/test1'
+    test "$(rbd mirror group snapshot schedule status -p rbd2/ns1 --group test1 --format xml |
+        xmlstarlet sel -t -v '//scheduled_groups/group/group')" = 'rbd2/ns1/test1'
+
+    rbd mirror group demote rbd2/ns1/test1
+    for i in `seq 12`; do
+        rbd mirror group snapshot schedule status | grep 'rbd2/ns1/test1' || break
+        sleep 10
+    done
+    rbd mirror group snapshot schedule status | expect_fail grep 'rbd2/ns1/test1'
+
+    rbd mirror group promote rbd2/ns1/test1
+    for i in `seq 12`; do
+        rbd mirror group snapshot schedule status | grep 'rbd2/ns1/test1' && break
+        sleep 10
+    done
+    rbd mirror group snapshot schedule status | grep 'rbd2/ns1/test1'
+
+    rbd mirror group snapshot schedule add 1h 00:15
+    test "$(rbd mirror group snapshot schedule ls)" = 'every 1h starting at 00:15:00'
+    rbd mirror group snapshot schedule ls -R | grep 'every 1h starting at 00:15:00'
+    rbd mirror group snapshot schedule ls -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2
+    rbd mirror group snapshot schedule ls -p rbd2 -R | grep 'every 1h starting at 00:15:00'
+    rbd mirror group snapshot schedule ls -p rbd2 -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    expect_fail rbd mirror group snapshot schedule ls -p rbd2/ns1
+    rbd mirror group snapshot schedule ls -p rbd2/ns1 -R | grep 'every 1h starting at 00:15:00'
+    rbd mirror group snapshot schedule ls -p rbd2/ns1 -R | grep 'rbd2 *ns1 *test1 *every 1m'
+    test "$(rbd mirror group snapshot schedule ls -p rbd2/ns1 --group test1)" = 'every 1m'
```

Comment:
Suggestion: remove a schedule and check that it is no longer present in the listing.
Does any scheduled snapshot also get removed?

Reply:
I've added this test in `test_mirror_group_snapshot_schedule()`